### PR TITLE
Improve escaping on the web interface

### PR DESCRIPTION
**Pull request by DL6ER**
# What does this implement/fix?

Fixes https://github.com/pi-hole/FTL/issues/1760, see https://github.com/pi-hole/FTL/issues/1760#issuecomment-1817486702 for additional context.

Before:
Screenshot from 2023-11-18 12-34-46

After:
Screenshot from 2023-11-18 12-48-03

**Related issue or feature (if applicable):** N/A

**Pull request in [docs](https://github.com/pi-hole/docs) with documentation (if applicable):** N/A

---
**By submitting this pull request, I confirm the following:** 

1. I have read and understood the [contributors guide](https://docs.pi-hole.net/guides/github/contributing/), as well as this entire template. I understand which branch to base my commits and Pull Requests against. 
2. I have commented my proposed changes within the code.
3. I am willing to help maintain this change if there are issues with it later.
4. It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
5. I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

## Checklist:

- [x] The code change is tested and works locally.
- [x] I based my code and PRs against the repositories `developmental` branch.
- [x] I [signed off](https://docs.pi-hole.net/guides/github/how-to-signoff/) all commits. Pi-hole enforces the [DCO](https://docs.pi-hole.net/guides/github/dco/) for all contributions
- [x] I [signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) all my commits. Pi-hole requires signatures to verify authorship
- [x] I have read the above and my PR is ready for review.

### CODE REVIEW

File: scripts/pi-hole/js/groups-domains.js [hunk 1]
```diff
@@ -503,16 +503,16 @@ function addDomain() {
     commentEl = $("#new_regex_comment");
   }
 
-  const comment = utils.escapeHtml(commentEl.val());
+  const comment = commentEl.val();
 
   // Check if the user wants to add multiple domains (space or newline separated)
   // If so, split the input and store it in an array
-  var domains = utils.escapeHtml(domainEl.val()).split(/\s+/);
+  var domains = domainEl.val().split(/\s+/);
```
Comment (DL6ER) [inline scripts/pi-hole/js/groups-domains.js:511]: We don't want to send HTML-escaped domains to FTL. We instead escape them below for the string that will be shown in the toast.

File: scripts/pi-hole/js/groups-domains.js [hunk 2]
```diff
@@ -503,16 +503,16 @@ function addDomain() {
     commentEl = $("#new_regex_comment");
   }
 
-  const comment = utils.escapeHtml(commentEl.val());
+  const comment = commentEl.val();
```
Comment (DL6ER) [inline scripts/pi-hole/js/groups-domains.js:507]: No need to escape the comment, otherwise it will end up escaped in the database.

Comment (yubiuser) [reply scripts/pi-hole/js/groups-domains.js:507]: We should also remove the coment escaping for clients, adlists, groups....

**Comment by yubiuser** [comment]
Interesting, that setting the content type works for us, but not for some users reported here: https://bugs.jquery.com/ticket/12326/

**Comment by DL6ER** [comment]
The small glitch I observed isn't a web issue. It will be fixed by https://github.com/pi-hole/FTL/pull/1763 so I'd say this PR is indeed ready for review.

---

> Interesting, that the setting the content type works for us, but not for some users reported here: https://bugs.jquery.com/ticket/12326/

These last comments you are referring to are **over 9 years** old, though. I guess the issue was fixed meanwhile but just nobody remembered this issue ticket (it is "closed as invalid").

**Review by rdwebdesign — DISMISSED** [review]

**Review by rdwebdesign — APPROVED** [review]

File: scripts/pi-hole/js/groups-common.js [hunk 3]
```diff
@@ -36,11 +36,6 @@ function processGroupResult(data, type, done, notDone) {
   // Loop over errors and display them
   data.processed.errors.forEach(function (error) {
     console.log(error); // eslint-disable-line no-console
-    utils.showAlert(
-      "error",
-      "",
-      `Error while ${notDone} ${type} ${utils.escapeHtml(error.item)}`,
-      error.error
-    );
+    utils.showAlert("error", "", `Error while ${notDone} ${type} ${error.item}`, error.error);
```
Comment (yubiuser) [inline scripts/pi-hole/js/groups-common.js:39]: Why did you remove it here? This is for showing ` ${error.item}`

File: scripts/pi-hole/js/groups-domains.js [hunk 4]
```diff
@@ -512,7 +512,7 @@ function addDomain() {
   domains = domains.filter(function (el) {
     return el !== "";
   });
-  const domainStr = utils.escapeHtml(JSON.stringify(domains));
+  const domainStr = JSON.stringify(domains);
```
Comment (yubiuser) [inline scripts/pi-hole/js/groups-domains.js]: I think this should stay escaped. You are reverting what you added with commit https://github.com/pi-hole/web/pull/2851/commits/bc1c6bc117eb88adb0c519b8ea9f24725ca755d3

File: scripts/pi-hole/js/groups-common.js (hunk 3 above)
Comment (DL6ER) [reply scripts/pi-hole/js/groups-common.js:39]: Because toasts themselves ensure that *everything* they display is escaped:

Screenshot from 2023-11-21 13-28-18

And double-escaping is not harmless, we'd get `"<"  ---escape--->  "&lt;"  ---escape--->  "&amp;lt;"`

File: scripts/pi-hole/js/groups-domains.js (hunk 4 above)
Comment (DL6ER) [reply scripts/pi-hole/js/groups-domains.js]: See my comment above, toasts now always escape themselves

**Review by yubiuser — APPROVED** [review]
